src/cloudsc_mix/CMakeLists.txt

```diff
@@ -0,0 +1,131 @@
+      set(CLOUDSC_GPU_OFFLOAD "ACC")
+    endif()
+    if( NOT ${CLOUDSC_GPU_OFFLOAD} MATCHES "ACC|OMP" )
+      message(FATAL_ERROR "CLOUDSC_GPU_OFFLOAD: '${CLOUDSC_GPU_OFFLOAD}' is not a valid option! (Allowed: 'ACC' and 'OMP')")
+    endif()
+
+    set(GPU_OFFLOAD ${${CLOUDSC_GPU_OFFLOAD}_OFFLOAD})
+
+    # Select kernel language
+    if( NOT DEFINED CLOUDSC_GPU_LANG )
+      set(CLOUDSC_GPU_LANG "CUDA")
+    endif()
+    if( NOT ${CLOUDSC_GPU_LANG} MATCHES "CUDA|HIP|SYCL" )
+      message(FATAL_ERROR "CLOUDSC_GPU_LANG: '${CLOUDSC_GPU_LANG}' is not a valid option! (Allowed: 'CUDA', 'HIP' and 'SYCL')")
+    endif()
+
+    set(GPU_LANG ${${CLOUDSC_GPU_LANG}_LANG})
+
+    if (CLOUDSC_GPU_LANG STREQUAL "CUDA")
+	enable_language(CUDA)
+	set(CMAKE_CUDA_FLAGS "-O3 -ffast-math")
+    elseif (CLOUDSC_GPU_LANG STREQUAL "HIP")
+	if(NOT DEFINED ROCM_PATH)
+          if(DEFINED ENV{ROCM_PATH})
+            set(ROCM_PATH $ENV{ROCM_PATH} CACHE PATH "Path to which ROCM has been installed")
+          else()
+            set(ROCM_PATH "/opt/rocm" CACHE PATH "Path to which ROCM has been installed")
+          endif()
+        endif()
+	enable_language(HIP)
+        find_package(hip REQUIRED)
+	set(CMAKE_HIP_FLAGS "${CMAKE_HIP_FLAGS} -O3 -ffast-math")
+    elseif (CLOUDSC_GPU_LANG STREQUAL "SYCL")
+	enable_language(CXX)
+        set(CMAKE_CXX_STANDARD 17)
+    endif()
+    
+    ecbuild_add_library(
+        TARGET dwarf-cloudsc-gpu-lib-${prec}
+        INSTALL_HEADERS LISTED
+	TYPE STATIC
+        SOURCES
+    	    yoecldp_c.h
+      	    cloudsc_c_k_caching.cpp
+            cloudsc_c_k_caching_mod.F90
+    	    PUBLIC_INCLUDES
+    	    	$<INSTALL_INTERFACE:include>
+            	$<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}>
+    	    PUBLIC_LIBS
+                $<${HAVE_OMP}:OpenMP::OpenMP_Fortran>
+                $<${HAVE_ACC}:OpenACC::OpenACC_Fortran>
+		cloudsc-common-${prec}-lib
+    )
+
+    # Provide definitions to target
+    foreach(_def CUDA_LANG HIP_LANG SYCL_LANG ACC_OFFLOAD OMP_OFFLOAD GPU_LANG GPU_OFFLOAD)
+      target_compile_definitions(dwarf-cloudsc-gpu-lib-${prec} PUBLIC ${_def}=${${_def}})
+    endforeach()
+
+    target_include_directories(dwarf-cloudsc-gpu-lib-${prec} PUBLIC $<INSTALL_INTERFACE:include> $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}>)
+
+    # CUDA specific
+    if (CLOUDSC_GPU_LANG STREQUAL "CUDA")
+      set_source_files_properties(cloudsc_c_k_caching.cpp PROPERTIES LANGUAGE CUDA)
+      target_include_directories(
+        dwarf-cloudsc-gpu-lib-${prec}
+        PUBLIC
+        ${CMAKE_CUDA_TOOLKIT_INCLUDE_DIRECTORIES}
+      )
+      if (NOT DEFINED CMAKE_CUDA_ARCHITECTURES)
+        target_compile_options(dwarf-cloudsc-gpu-lib-${prec} PRIVATE $<COMPILE_LANGUAGE:CUDA>)
+      else()
+        target_compile_options(dwarf-cloudsc-gpu-lib-${prec} PRIVATE $<$<COMPILE_LANGUAGE:CUDA>:
+	  -gencode arch=compute_${CMAKE_CUDA_ARCHITECTURES},code=sm_${CMAKE_CUDA_ARCHITECTURES}>)
+      endif()
```

Comment:
Can't we use `$<${HAVE_CUDA}:${CLOUDSC_CUDA_FLAGS}>` in ecbuild_add_library instead?
https://github.com/ecmwf-ifs/dwarf-p-cloudsc/blob/70cddc4101fa726b1921a4230804a342646fee67/cmake/features/CUDA.cmake#L13-L17